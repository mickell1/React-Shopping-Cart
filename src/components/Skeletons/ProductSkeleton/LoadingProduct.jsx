import React from "react";
import '../../Products/Products.scss'

const LoadingProduct = () => {
  return (
    <div className="product loading">
      <div className="product-image" />
      <div className="product-text" />
      <div className="product-text" />
      <div className="product-text" />
      <div className="product-button" />
    </div>
  );
};

export default LoadingProduct;
