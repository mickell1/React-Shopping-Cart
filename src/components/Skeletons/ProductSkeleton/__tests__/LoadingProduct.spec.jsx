import React from 'react';
import { shallow, mount } from 'enzyme';
import LoadingProduct from '../LoadingProduct';

describe('LoadingProduct', () => {
  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<LoadingProduct /> )  
  })

  it('should render the LoadingProduct Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });

  it('should render div with classname', () => {
    wrapper = mount(<LoadingProduct />)
    expect(wrapper.find('div').first().hasClass('product loading')).toEqual(true);
  })
});