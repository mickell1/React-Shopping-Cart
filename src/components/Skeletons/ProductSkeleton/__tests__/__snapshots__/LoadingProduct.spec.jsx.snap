// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingProduct should render the LoadingProduct Component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingProduct />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="product-image"
        />,
        <div
          className="product-text"
        />,
        <div
          className="product-text"
        />,
        <div
          className="product-text"
        />,
        <div
          className="product-button"
        />,
      ],
      "className": "product loading",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-image",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-text",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-button",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="product-image"
          />,
          <div
            className="product-text"
          />,
          <div
            className="product-text"
          />,
          <div
            className="product-text"
          />,
          <div
            className="product-button"
          />,
        ],
        "className": "product loading",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-image",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-text",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-button",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
